src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NullTestAroundInstanceOfCheck.java

```diff
@@ -0,0 +1,178 @@
+            final DetailAST instanceOfNode =
+                    landNode.findFirstToken(TokenTypes.LITERAL_INSTANCEOF);
+            if (notEqualNode != null && instanceOfNode != null
+                    && notEqualNode.findFirstToken(TokenTypes.LITERAL_NULL) != null) {
+                final DetailAST dotNode1 = getDotNode(notEqualNode);
+                final DetailAST dotNode2 = getDotNode(instanceOfNode);
+                final String string1;
+                if (dotNode1 == null) {
+                    string1 = getIdentNode(notEqualNode).getText();
+                }
+                else {
+                    final DetailAST identNode1 = getIdentNode(dotNode1);
+                    string1 = getText(identNode1);
+                }
+                final String string2;
+                if (dotNode2 == null) {
+                    string2 = getIdentNode(instanceOfNode).getText();
+                }
+                else {
+                    final DetailAST identNode2 = getIdentNode(dotNode2);
+                    string2 = getText(identNode2);
+                }
+                if (string1.equals(string2)) {
+                    log(ast, MSG_KEY);
+                }
+            }
+        }
+    }
+
+    private static String getText(DetailAST ast) {
+        return ast.getText() + ast.getNextSibling().getText();
+    }
+
+    private static DetailAST getDotNode(DetailAST ast) {
+        return ast.findFirstToken(TokenTypes.DOT);
+    }
+
+    private static DetailAST getIdentNode(DetailAST ast) {
+        return ast.findFirstToken(TokenTypes.IDENT);
+    }
+
+    /**
+     * Check if there is a unnecessary 'null' check before 'instanceof' expression when
+     * 'instanceof' expression is on next line of 'null' check.
+     *
+     * @param ast DetailAST ast
+     * @return error boolean
+     */
+    private static boolean checkOnNextLine(DetailAST ast) {
+        final DetailAST notEqualNode = ast.findFirstToken(TokenTypes.EXPR)
+                .findFirstToken(TokenTypes.NOT_EQUAL);
+        boolean error = false;
+        if (notEqualNode != null
+                && notEqualNode.findFirstToken(TokenTypes.LITERAL_NULL) != null) {
+            final DetailAST slistNode = ast.findFirstToken(TokenTypes.SLIST);
+            final DetailAST nestedIfNode;
+            if (slistNode == null) {
+                nestedIfNode = ast.findFirstToken(TokenTypes.LITERAL_IF);
+            }
+            else {
+                nestedIfNode = slistNode.findFirstToken(TokenTypes.LITERAL_IF);
+            }
+            if (nestedIfNode != null) {
+                final DetailAST instanceOfNode =
+                        nestedIfNode.findFirstToken(TokenTypes.EXPR)
+                                .findFirstToken(TokenTypes.LITERAL_INSTANCEOF);
+                if (instanceOfNode != null
+                        && getIdentNode(notEqualNode).getText()
+                        .equals(getIdentNode(instanceOfNode).getText())) {
```

Comment:
and also here you need to check class fields `person.name`

Reply:
done